prost-types/src/any_v2.rs

```diff
@@ -0,0 +1,881 @@
+use core::{any::Any as CoreAny, cell::RefCell, fmt::Debug};
+use std::{
+    collections::HashMap,
+    sync::{Arc, OnceLock, RwLock},
+};
+
+use ::serde as _serde;
+pub use prost::serde::private::JsonValue;
+use prost::{
+    bytes::{Buf, BufMut},
+    serde::SerdeMessage,
+    Message, Name,
+};
+
+use crate::smallbox::{smallbox, SmallBox};
+
+mod private {
+    pub trait Sealed {}
+}
+
+pub trait AnyValue: CoreAny + Message + private::Sealed {
```

Comment:
Can you explain what any v2 is and why it is better?

Reply:
This basically tries to address the "improved Any" bulletpoint in #624. The one Any implementation provides support for JSON and "cross" serialization (JSON <-> Binary).

The summary for that change is basically going from:

```rust
struct OldAny {
    type_path: String,
    data: Vec<u8>,
}
```

to

```rust
struct NewAny {
    type_path: String,
    kind: enum {
        Protobuf(Vec<u8>),
        Json(JsonValue),
        Dyn(Box<dyn AnyValue>),
    }
}
```